src/link/resolvePeers.ts

```diff
@@ -185,7 +179,8 @@ function resolvePeersOfNode (
       hasBundledDependencies: node.pkg.hasBundledDependencies,
       fetchingFiles: node.pkg.fetchingFiles,
       resolution: node.pkg.resolution,
-      centralLocation,
+      centralLocation: node.pkg.cacheByNodeVersion || centralLocation,
```

Comment:
You'll have to change it here: https://github.com/pnpm/supi/blob/b8bc8cf8e30eab886bc5b2fd1c2bfa7395efa036/src/link/resolvePeers.ts#L172

centralLocation  is used for peripheralLocation sometimes, so it should have the correct value

Reply:
I don't understand this sentence